### PR TITLE
fixes: #34 list_orders takes params

### PR DESCRIPTION
**Pull request by h55nick**
allow `list_orders` to take a `params` dict that will be passed through to `.get` as query parameters. 

Note: No validation of valid params here but should be done at the API level

### CODE REVIEW

**Comment by h55nick** [comment]
also considering changing:
```
        params['status'] = params['status'] or status or None #might not be very pythonic
        resp = self.get('/orders', params)
        return [Order(o) for o in resp]
```
to simplify

**Comment by umitanuki** [comment]
Thanks for the PR. I can see bundling parameters into `param` may be clean, but for now I would rather keep them separate for reasons: 1) to be consistent with other methods, and 2) documentation/explanation purpose (you can do rest.list_orders? in ipython to see parameter names quickly)

File: alpaca_trade_api/rest.py [hunk 1]
```diff
@@ -130,9 +130,11 @@ def get_account(self):
         resp = self.get('/account')
         return Account(resp)
 
-    def list_orders(self, status=None):
-        '''Get a list of orders'''
-        params = dict()
+    def list_orders(self, params=dict(), status=None):
```
Comment (umitanuki) [inline alpaca_trade_api/rest.py]: FYI, `dict()` is initialized and reused this way, which means the second invocation will have the same dict instance with keys filled as in the first call.
https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html

Comment (h55nick) [reply alpaca_trade_api/rest.py]: fascinating, certainly explains it well and curious why I was never aware! Can change once we decide on the direction.

**Comment by h55nick** [comment]
@umitanuki understand the desire to be clean of course. If you have an example of passing through parameters I'd be happy to make the update. One of my goals here would be not to require consistent updates to this repo for each api update so instead of having it strictly pass-through vs containing logic. This is why I would like to actively avoid adding parameters as 'status' is used now (kept status so we didn't break backward compatibility). 

Many thanks for your comments! Let me know what you are thinking.

**Comment by h55nick** [comment]
@umitanuki I've updated the PR according to your comments but leaving the option to pass in parameters like done previously. Let me know what you think :-)

**Comment by umitanuki** [comment]
Looks good to me. Thanks for the work!

**Pull request merged**